src/lib/provider/withGlobalProps.jsx

```diff
@@ -1,21 +1,38 @@
+import PropTypes from 'prop-types';
 import React from 'react';
 import RUIContext from './RUIContext';
 
 export default (Component, componentName) => {
-  const WithGlobalPropsComponent = (props) => (
+  const WithGlobalPropsComponent = ({
+    forwardedRef,
+    ...rest
+  }) => (
     <RUIContext.Consumer>
       {({ globalProps }) => {
         const contextGlobalProps = globalProps ? globalProps[componentName] : null;
 
         return (
           <Component
+            ref={forwardedRef}
```

Comment:
`ref={forwardedRef}` should be placed after `restProps`

Reply:
I'm not sure.:

1. In https://reactjs.org/docs/forwarding-refs.html they have it before `...rest`.
2. The only potential unwanted override could be  if the `props` contained a `ref` prop along with the `forwardedRef` prop. i don't think that is technically possible because of this line: `return React.forwardRef((props, ref) => (<WithGlobalPropsComponent {...props} forwardedRef={ref} />));`

Reply:
@mbohal It's not true. They do not have any recommendation about the order. In some examples, they have it before rest props, in another after props. They don't give a damn about the order.

But_
1) We agreed on Monday that we want to have first rest props, then "local" props, so we cannot accidentally override it. 
2) In components, we always use `transferProps` which truncates those unwanted props from the object. This is not standard component, so I would preserve order I suggest and which we use in the rest of components. I am not sure whether to use transferProps in this HoC, I might leave it there without but it is up to you.

Reply:
I have no strong feelings about it, I just wanted to point to the example where it was before.

>  I might leave it there without

I agree